transport/bdp_estimator.go

```diff
@@ -0,0 +1,106 @@
+package transport
+
+import (
+	"sync"
+	"time"
+)
+
+const (
+	limit = (1 << 20) * 4
+	alpha = 0.9
+)
+
+var (
+	bdpPing = &ping{data: [8]byte{1, 2, 3, 4, 5, 6, 7, 8}}
+)
+
+type bdpEstimator struct {
+	side string
+
+	mu                sync.Mutex
+	bdp               uint32
+	sample            uint32    // Current bdp sample.
+	sentAt            time.Time // Time when the ping was sent.
+	bwMax             float64
+	isSent            bool
```

Comment:
We don't take a timestamp until the ping is actually written on the wire. However, we do want to calculate all the bytes received from the moment we decided to make a measurement (including the bytes that prompted this new measurement cycle).